<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Start Webots -->
    <arg name="no-gui" default="false" doc="Start Webots with minimal GUI"/>
    <arg name="world" default="square" doc="The name of the webots world to launch" />
    <include file="$(find simulation)/launch/webots.launch">
        <arg name="mode" value="pause"/>
        <arg name="no-gui" value="$(arg no-gui)"/>
        <!-- modify with path to world-->
        <arg name="world" value="$(find simulation)/worlds/$(arg world).wbt"/>
    </include>

    <!-- External Webots Controller -->
    <arg name="keyboard" default="0" doc="The type of keyboard controller to use" />
    <node name='sim_controller' pkg='simulation' type='full_controller' output='screen' respawn='true'>
        <param name="use_keyboard_control" value="$(eval arg('keyboard') == 0)" />
        <param name="keyboard/lvel_scale" value="1" />
    </node>

    <!-- External Keyboard Controllers -->
    <group if="$(eval arg('keyboard') != 0)">
        <!-- C++ Keyboard Controller -->
        <node if="$(eval arg('keyboard') == 1)" name="keyboard_controller" pkg="controllers" type="keyboard_controller" output="screen">
            <param name="avel_scale" value="0.3"/>
            <param name="lvel_scale" value="0.3"/>
        </node>

        <!-- Python Keyboard Controller -->
        <node if="$(eval arg('keyboard') == 2)" name="keyboard_controller" pkg="controllers" type="KeyboardController.py" output="screen">
            <param name="avel_scale" value="0.3"/>
            <param name="lvel_scale" value="0.3"/>
        </node>
    </group>
</launch>
